src/mlpack/methods/cf/cf_impl.hpp

```diff
@@ -69,37 +37,59 @@ CF::CF(const arma::mat& data,
     // Set default value of 5.
     this->numUsersForSimilarity = 5;
   }
-
-  Train(data, factorizer);
 }
 
 /**
- * Construct the CF object using an instantiated factorizer.
+ * Construct the CF object using an instantiated decomposition policy.
  */
-template<typename FactorizerType>
-CF::CF(const arma::sp_mat& data,
-       FactorizerType factorizer,
+template<typename DecompositionPolicy>
+template<typename mat_type>
+CF<DecompositionPolicy>::CF(mat_type const& data,
+       const DecompositionPolicy& decomposition,
        const size_t numUsersForSimilarity,
        const size_t rank,
-       const typename std::enable_if_t<
-           !FactorizerTraits<FactorizerType>::UsesCoordinateList>*) :
+       const size_t maxIterations,
+       const double minResidue,
+       const bool mit) :
+    decomposition(decomposition),
     numUsersForSimilarity(numUsersForSimilarity),
     rank(rank)
 {
   // Validate neighbourhood size.
   if (numUsersForSimilarity < 1)
   {
-    Log::Warn << "CF::CF(): neighbourhood size should be > 0("
+    Log::Warn << "CF::CF(): neighbourhood size should be > 0 ("
         << numUsersForSimilarity << " given). Setting value to 5.\n";
-    // Setting Default Value of 5
+    // Set default value of 5.
     this->numUsersForSimilarity = 5;
   }
 
-  Train(data, factorizer);
+  Train(data, decomposition, maxIterations, minResidue, mit);
+}
+
+// Apply the decomposition policy.
+template<typename DecompositionPolicy>
+template<typename mat_type>
+void CF<DecompositionPolicy>::Apply(mat_type const& data,
+            const arma::sp_mat& cleanedData,
+            const size_t rank,
+            arma::mat& w,
+            arma::mat& h,
+            const size_t maxIterations,
+            const double minResidue,
+            const bool mit)
+{
+  decomposition.Apply(data, cleanedData, rank, w,
+                      h, maxIterations, minResidue, mit);
```

Comment:
The next line should be tabbed twice from where the previous line began, see https://github.com/mlpack/mlpack/wiki/DesignGuidelines#naming-conventions for more information.